src/game-loop/src/main-dude/states/MainDudeBaseState.cpp

```diff
@@ -0,0 +1,51 @@
+#include "main-dude/states/MainDudeBaseState.hpp"
+#include "main-dude/MainDude.hpp"
+#include "Input.hpp"
+
+void MainDudeBaseState::handle_movement(MainDude& main_dude, const Input& input, float speed)
+{
+    if (input.left())
```

Comment:
The whole idea of having main dude split into states is to contain logics in there.

Reply:
Including handling input.

Reply:
All right, I will drop the subject of input refactoring _for now_